app/views/api_keys/new.html.erb

```diff
@@ -15,6 +15,12 @@
       <% end %>
     </div>
 
+    <div class="form__group">
+      <%= label_tag :rubygem, t(".rubygem_scope"), class: "form__label" %>
+      <p><%= t(".rubygem_scope_info") %></p>
+      <%= f.text_field :rubygem_name, value: @api_key.rubygem&.name, placeholder: t(".all_gems"), class: "form__input" %>
```

Comment:
> >just a textbox. Prone to typos
>
> I think we should start with this. We would show validation errors for typos.

I got a textbox working for this with a case for typos and a case where the owner doesn't own the gem specified. I put that logic in the model so I could have all the validations in one place. It's in 2 commits atm 8fbbfa68aef060db933856ddb79b6799082617cb ce0c70659e37fa8fd0afdf7a3ae40fb6b5dabce3.

 I tried having the logic in controller also (it's commented out), but I had to early return and display the typo message before other validations since calling save effectively regenerates and clears previous errors. I'm keen to hear if anyone has other ideas.

<details>
<summary>Create API key</summary>
<img width="883" alt="Screen Shot 2022-02-07 at 9 30 58 PM" src="https://user-images.githubusercontent.com/42748004/152907590-20a54598-755f-44fc-8741-a73220a90e52.png">
<img width="917" alt="Screen Shot 2022-02-07 at 9 29 59 PM" src="https://user-images.githubusercontent.com/42748004/152907600-154461f6-9ad3-41b1-a9b3-003681c27dd3.png">
</details>

Reply:
have you checked how pypi is handling this?

Reply:
sure. we can go back to dropdown.